app/javascript/article-form/articleForm.jsx

```diff
@@ -251,11 +251,12 @@ export default class ArticleForm extends Component {
 
   handleArticleError = (response, publishFailed = false) => {
     window.scrollTo(0, 0);
+    const { published } = this.state;
     this.setState({
       errors: response,
       submitting: false,
       // Even if it's an update that failed, published will still be set to true
```

Comment:
To be honest, I wasn't quite sure about the rationale around this comment. Should this function touch the `published` state at all?

This function is used as error callback (`errorCb`) in `submitArticle` (defined in (`app/javascript/article-form/actions.js`):

```js
// If there is an error and the method is POST, we know they are trying to publish.
errorCb(response, method === 'POST');
```

I do somewhat get the idea behind this, but it (a) doesn't seem to properly take drafts into account and (b) intuitively feels wrong since the article won't in fact be published if it fails validation, even if we set this attribute.

Reply:
I agree on both counts. That logic should probably be revisited in another PR.

Also looking at the publish logic

https://github.com/forem/forem/blob/c2c1366a4ddc452f9d7e587d1b930391acbe0c06/app/javascript/article-form/articleForm.jsx#L207-L213

`setState` is not immediate and it's bad practice to mutate state like so `state.published` as there are no guarantees. That should probably be reworked as well.

Reply:
@nickytonline If you don't mind, I'll leave further refactoring and fixes up to you, as this is more your domain than mine 😄 Maybe add an issue for it so it finds its way into the next planning meeting.

Reply:
I created a separate issue, #/12349.